lxd/main_nsexec.go

```diff
@@ -117,22 +90,22 @@ int dosetns(int pid, char *nstype) {
 
 void attach_userns(int pid) {
 	char nspath[PATH_MAX];
-	char userns_source[PATH_MAX];
-	char userns_target[PATH_MAX];
+	char userns_source[17];
+	char userns_target[17];
```

Comment:
We should hardcode a random number when using a fixed-size buffer. I have no idea what that path is supposed to be sufficient for without reading through the whole code until the point where it's first used. At first I thought this was `/proc/<pid>/ns/<ns_name>` instead of the contents of that path. Anyway, for `nspath` we can use this like we do in LXC::
```
 /* 
  * strlen("/proc") = 5
  * +
  * 64 bit int to chars = 2^64 - 1 = 21 chars
  * +
  * strlen("/ns") = 3
  * +
  * strlen("/<ns_name>") = 20
  * +
  * \0 = 1
  */
#define __NS_PATH_LEN 50
char path[__NS_PATH_LEN];
```

and for target - assuming this is the inode number:

```
/* 64 bit int to chars = 2^64 - 1 = 21 chars + \0 */
#define  __NS_PATH_CONTENTS 21
```

Reply:
Both of them are inode numbers.

Reply:
What's annoying is that as readlink isn't NULL terminated, we can't just use strcmp on it, we were doing that before and it was a bug getting us false positives in the check.

Reply:
Fixed to use 22 chars length and manual null termination with strcmp.